### PR TITLE
Fixes and enhancements for .22

### PR DESCRIPTION
**Pull request by JustinFreitas**
Deprecation fixes, token height/name fixes, copy to chat functionality support, OSE fixes, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,15 +72,6 @@ This theme uses a very high loadorder of 1500 in order to be able to overwrite m
...
 
-### Want to buy me a coffee?
-[PayPal War Chest](https://www.paypal.com/paypalme/DnDKriegskasse)
-or
-[Ko.Fi](https://ko-fi.com/sirmotte)
-
-### Want to buy me a DMs Guild gift instead?
-Here's my [Dungeon Masters Guild Maps Wishlist](https://www.dmsguild.com/wishlist_public.php?public_id=2593421&list_id=3816273&buying_for=sirmotte%40posteo.de&quot;]Wishlist%20for%20Maps), or my general [PDF Wishlist](https://www.dmsguild.com/wishlist_public.php?public_id=2593421&list_id=3869339&buying_for=sirmotte%40posteo.de)
-
-
```
Comment (JustinFreitas) [inline README.md:84]: Looks like you did some readme cleanup... if not your changes, just alter my PR.

File: extension.xml [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
 
 <root version="3.2">
-	<announcement text="Theme Loaded: \rSir Motte's Magnificent Darkness\rVersion: 2.9.21 - Knightowl" font="oocfont" icon="footface" />
+	<announcement text="Theme Loaded: \rSir Motte's Magnificent Darkness\rVersion: 2.9.22 - I See It Now" font="oocfont" icon="footface" />
 
 	<properties>
 		<name>Theme: Hearth</name>
 		<author>Sir Motte - Co. Authors: MeAndUnique and Saagael / Contributions by bmos, Zarestia and JustinFreitas.</author>
-		<version>2.9.21</version>
+		<version>2.9.22</version>
```
Comment (JustinFreitas) [inline extension.xml:10]: Versioning stuff... change as necessary.

File: graphics/graphics_fonts.xml [hunk 3]
```diff
@@ -98,6 +98,10 @@
 		<ttf file="graphics/fonts/Roboto-Regular.ttf" name="Roboto" size="16" />
 		<color value="#77CE77" /> <!-- Replaceable Color: Chat Font 4 -->
 	</font>
+    <font name="chat_record_share">
+		<ttf file="graphics/fonts/Roboto-Bold.ttf" name="Roboto Bold" size="16" />
+		<color value="#DDDDDD" />
+	</font>
```
Comment (JustinFreitas) [inline graphics/graphics_fonts.xml:104]: For the new chat record share functionality.

File: graphics/graphics_frames.xml [hunk 4]
```diff
@@ -261,20 +261,24 @@
...
 	</framedef>
 	<framedef name="chatframe_story">
 		<bitmap file="graphics/pngs/chatframe_story.png" />
-		<offset>18,18,17,17</offset>
+		<offset>7,7,7,7</offset>
 	</framedef>
 	<framedef name="chatframe_whisper">
 		<bitmap file="graphics/pngs/chatframe_whisper.png" />
-		<offset>17,17,7,7</offset>
+		<offset>7,7,7,7</offset>
```
Comment (JustinFreitas) [inline graphics/graphics_frames.xml:272]: Fixes for chat, story, and whisper frame formatting to maximize useable space.

File: graphics/graphics_frames.xml [hunk 5]
```diff
@@ -261,20 +261,24 @@
...
 	</framedef>
 	<framedef name="diebox">
 		<bitmap file="graphics/pngs/diebox.png" />
 		<offset>16,16,16,16</offset>
 	</framedef>
+	<framedef name="chatframe_record_share">
+		<bitmap file="graphics/pngs/chatframe_record_share.png" />
+		<offset>8,40,8,8</offset>
+	</framedef>
```
Comment (JustinFreitas) [inline graphics/graphics_frames.xml:281]: Image for the new chat record share functionality.

File: graphics/graphics_frames.xml [hunk 6]
```diff
@@ -754,15 +758,15 @@
...
+		<bitmap file="graphics/pngs/mini_name_height.png" />
 		<offset>7,7,7,7</offset>
 	</framedef>
 	<framedef name="token_ordinal">
 		<bitmap file="graphics/pngs/tokennumber.png" />
 		<offset>6,6,6,6</offset>
 	</framedef>
 	<framedef name="token_height">
-		<bitmap file="graphics/pngs/mini_name.png" />
+		<bitmap file="graphics/pngs/mini_name_height.png" />
```
Comment (JustinFreitas) [inline graphics/graphics_frames.xml:769]: These two are temporary fixes for the token height and name problems with the latest version of FGU.

File: graphics/graphics_icons.xml [hunk 7]
```diff
@@ -8,7 +8,7 @@
 
 	<!-- CoreRPG -->
 	<!-- Portrait -->
-	<icon name="charlist_base" file="graphics/pngs/portrait.png" />
+	<icon name="charlist_base" file="graphics/pngs/portrait_token.png" />
```
Comment (JustinFreitas) [inline graphics/graphics_icons.xml:11]: Fix for the missing gold frame around PC portraits (the left one in 5e... right one was always ok).

File: graphics/graphics_icons.xml [hunk 8]
```diff
@@ -735,4 +736,11 @@
 	<icon name="task8" file="graphics/pngs/task8.png" />
 	<icon name="task9" file="graphics/pngs/task9.png" />
 	<icon name="task10" file="graphics/pngs/task10.png" />
+
+    <!-- Copy to Chat, Used in Reference, etc. -->
+    <icon name="refresh" file="graphics/pngs/refresh.png" />
+	<icon name="content_chat" file="graphics/pngs/content_chat.png" />
+	<icon name="content_copy" file="graphics/pngs/content_copy.png" />
+	<icon name="content_paste" file="graphics/pngs/content_paste.png" />
```
Comment (JustinFreitas) [inline graphics/graphics_icons.xml:744]: For the new copy to chat functionality.

File: rulesets/dnd4e_compilation.xml [hunk 9]
```diff
@@ -1,6 +1,18 @@
...
+			function onInit()
+				if super and super.onInit then
+					super.onInit();
+				end
+				if hptemp then
+					hptemp.setColor(ColorManager.COLOR_TEMP_HP);
+				end
+			end
+		</script>
+	</windowclass>
```
Comment (JustinFreitas) [inline rulesets/dnd4e_compilation.xml:15]: Moved here from CoreRPG, where it was misplaced.

File: rulesets/corerpg_compilation.xml [hunk 10]
```diff
@@ -458,6 +478,75 @@
...
+	</template>
+	<template name="button_collapse">
+		<button_icon>
+			<icon normal="button_collapse_w" />
+			<tooltip textres="tooltip_collapse" />
+			<target>grouplist,list</target>
+			<collapse />
+			<script file="common/scripts/button_expand_collapse.lua" />
+		</button_icon>
+	</template>
```
Comment (JustinFreitas) [inline rulesets/corerpg_compilation.xml:549]: Reference manual copy to chat functionality.

File: rulesets/corerpg_compilation.xml [hunk 11]
```diff
@@ -626,54 +715,27 @@
...
-				if super and super.onInit then
-					super.onInit();
-				end
-				if dying then
-				dying.setColor(ColorManager.COLOR_HEALTH_CRIT_WOUNDS);
-				end
-			end
-		</script>
-	</windowclass>
-</root>
```
Comment (JustinFreitas) [inline rulesets/corerpg_compilation.xml:679]: Incorrect location.  Moved to the various ruleset XML files.

File: rulesets/scripts/manager_theme.lua [hunk 12]
```diff
@@ -2,8 +2,8 @@
 isThemeLoaded = false;
 
 function onInit()
-	DesktopManager.setSidebarDockCategoryIconColor("FFF0CA");
-	DesktopManager.setSidebarDockIconColor("FFF0CA");
+	ColorManager.setSidebarCategoryIconColor("FFF0CA");
+	ColorManager.setSidebarRecordIconColor("FFF0CA");
```
Comment (JustinFreitas) [inline rulesets/scripts/manager_theme.lua:6]: Deprecation fixes.

File: scripts/manager_color_2.lua [hunk 13]
```diff
@@ -55,8 +55,6 @@ function onInit()
 
 	ColorManager.COLOR_MODIFIED_VALUE = "F3D876"; -- Replacable Color: Modified VALUE
 
-	ColorManager.setButtonIconColor(ColorManager.COLOR_SECONDARY_FOREGROUND);
-
```
Comment (JustinFreitas) [inline scripts/manager_color_2.lua]: Unnecessary call, removed.

File: scripts/manager_desktop2.lua [hunk 14]
```diff
@@ -9,8 +9,8 @@ local _cDockIconColor = "DDDDDD"; -- Replaceable Color: Secondary Foreground
...
-	DesktopManager.setSidebarDockCategoryTextColor(_cDockCategoryTextColor);
-	DesktopManager.setSidebarDockIconColor(_cDockIconColor);
-	DesktopManager.setSidebarDockTextColor(_cDockTextColor);
-end
\ No newline at end of file
+	ColorManager.setSidebarCategoryIconColor(_cDockCategoryIconColor);
+	ColorManager.setSidebarCategoryTextColor(_cDockCategoryTextColor);
+	ColorManager.setSidebarRecordIconColor(_cDockIconColor);
+	ColorManager.setSidebarRecordTextColor(_cDockTextColor);
+end
```
Comment (JustinFreitas) [inline scripts/manager_desktop2.lua:16]: Deprecation fixes.

**Comment by JustinFreitas** [comment]
Hi SirMotte.  I have a build script that I use (it's for Windows) to quickly build the extension.  If you would like me to check that in also, just say.  Thanks, Justin

**Comment by SirMotte** [comment]
Something that eases workload? Go for it! :)

**Pull request merged**